paccmann_predictor/models/bimodal_mca.py

```diff
@@ -250,25 +317,31 @@ def __init__(self, params, *args, **kwargs):
 
     def forward(self, smiles, proteins, confidence=False):
         """Forward pass through the biomodal MCA.
-
         Args:
             smiles (torch.Tensor): of type int and shape
                 `[bs, smiles_padding_length]`.
             proteins (torch.Tensor): of type int and shape
                 `[bs, protein_padding_length]`.
             confidence (bool, optional) whether the confidence estimates are
                 performed.
-
         Returns:
             (torch.Tensor, torch.Tensor): predictions, prediction_dict
-
             predictions is IC50 drug sensitivity prediction of shape `[bs, 1]`.
             prediction_dict includes the prediction and attention weights.
         """
 
         # Embedding
-        embedded_smiles = self.smiles_embedding(smiles.to(torch.int64))
-        embedded_protein = self.protein_embedding(proteins.to(torch.int64))
+        if self.peptide_embedding == 'smiles':
+            embedded_smiles = self.smiles_embedding(smiles.to(torch.int64))
+        elif self.encoding == 'learned':
+            embedded_smiles = self.protein_embedding(smiles.to(torch.int64))
+        else:
+            embedded_smiles = smiles.to(torch.float)
+
+        if self.encoding == 'learned':
+            embedded_protein = self.protein_embedding(proteins.to(torch.int64))
+        else:
+            embedded_protein = proteins.to(torch.float)
```

Comment:
This needs some improvement. First, we should, if possible, avoid if/else constructions in the forward pass since they're executed with every batch. Here, it doesnt seem necessary. Also the if/elif construction queries entirely independent things. Plus: Why should the SMILES be embedded with protein_embeddings?? 

I think this bit needs some refinement and it should then be tested that the code runs with different configs.